social media icons on mentor page

ePSA-eJya opened the pull request:
## Related Issue

<!--
added social media icons on mentor page
-->

Closes: #221

## Description of Changes

<!--
Clearly and concisely describe the modifications made to successfully resolve the assigned issue. Include any pertinent information about new files or any other relevant details.

For example:
- Added a new function to handle XYZ.
- Updated the ABC module to fix the bug.
- Refactored code in the DEF class for improved performance.
-->

## Checklist:

<!--
Mark the checkboxes to indicate completion. Example:
- [x] My code follows the style guidelines of this project.
-->

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [x] I have included comments in areas that may be difficult to understand.
- [x] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.

added social media icons
![image](https://github.com/anmode/grabtern-frontend/assets/120899038/4e92e42b-9fec-465d-b2b5-ef9fbda06d11)

ePSA-eJya commented:
please review and merge the pr @anmode

anmode commented:
it is good! but I cannot assign any level on 2 or 3 lines of code....let's optimize further.
1. add discord link
2. Hover effect on footer links is not looking good please optimize
3. Create a seperate file for footer.css https://github.com/anmode/grabtern-frontend/pull/224 for refer

ePSA-eJya commented:
> it is good! but I cannot assign any level on 2 or 3 lines of code....let's optimize further.
> 
> 1. add discord link
> 2. Hover effect on footer links is not looking good please optimize
> 3. Create a seperate file for footer.css [Break the changes to modular.css #224](https://github.com/anmode/grabtern-frontend/pull/224) for refer
> 
> if you don't wanna this furthur i can merge but i can't assign any level... Main thing is learning I hope you will understand. I hope you continue refering these three points. Thank you

I would want to work on this.

anmode commented:
great to hear! For now you continue.. I'll provide discord furthur you can add that afterward. If any issue feel free to ask here or discord(gssoc)

anmode commented:
https://github.com/anmode/grabtern-frontend/issues/242

ePSA-eJya commented:
> #242

Does it have anything to do with me? @anmode

ePSA-eJya commented:
do you want me to create a separate css file for footer?
@anmode

anmode commented:
> > #242
> 
> Does it have anything to do with me? @anmode

yess as you are modifying the footer.. this would be a small line change please have a look! and yess please create a seperate css file

anmode commented:
> i have added some hover effects on social media links. ![image](https://user-images.githubusercontent.com/120899038/243088801-1d01505f-120c-499f-8dd7-2cd7176f68e8.png) ![image](https://user-images.githubusercontent.com/120899038/243088880-6a907ea6-9d1a-4acd-acb4-9068883fd853.png) also i have changed hover effectson the links to make it more prominent ![image](https://user-images.githubusercontent.com/120899038/243089570-48179c52-3d42-496e-b358-f1354daba8d3.png)
> 
> is it fine? @anmode or do you want some more changes?

hover effect on links not looks good!

ePSA-eJya commented:
> > i have added some hover effects on social media links. ![image](https://user-images.githubusercontent.com/120899038/243088801-1d01505f-120c-499f-8dd7-2cd7176f68e8.png) ![image](https://user-images.githubusercontent.com/120899038/243088880-6a907ea6-9d1a-4acd-acb4-9068883fd853.png) also i have changed hover effectson the links to make it more prominent ![image](https://user-images.githubusercontent.com/120899038/243089570-48179c52-3d42-496e-b358-f1354daba8d3.png)
> > is it fine? @anmode or do you want some more changes?
> 
> hover effect on links not looks good!

are the effects on social media icons good? or should i modify them as well? @anmode

anmode commented:
> > > i have added some hover effects on social media links. ![image](https://user-images.githubusercontent.com/120899038/243088801-1d01505f-120c-499f-8dd7-2cd7176f68e8.png) ![image](https://user-images.githubusercontent.com/120899038/243088880-6a907ea6-9d1a-4acd-acb4-9068883fd853.png) also i have changed hover effectson the links to make it more prominent ![image](https://user-images.githubusercontent.com/120899038/243089570-48179c52-3d42-496e-b358-f1354daba8d3.png)
> > > is it fine? @anmode or do you want some more changes?
> > 
> > 
> > hover effect on links not looks good!
> 
> are the effects on social media icons good? or should i modify them as well? @anmode

they are good!

ePSA-eJya commented:
if you could tell me what hover effects you expect on the links it would be easier for me work on them @anmode

anmode commented:
see footer: https://mentro.tech/ I wish we can have our hover color which match to our site

ePSA-eJya commented:
> see footer: https://mentro.tech/ I wish we can have our hover color which match to our site

ok i will do that

ePSA-eJya commented:
![image](https://github.com/anmode/grabtern-frontend/assets/120899038/84f490a2-a5fc-4e48-9cfa-124faef7c53b)

i have tried inverting the color scheme in the footer section as the previous color was barring the visibility.
please review @anmode

anmode commented:
oh no no!! okay then do one thing.. remove hover from hyperlink just make on hover a under line comes in......just add one more social that is discord.
and other things create css file differently

ePSA-eJya commented:
![image](https://github.com/anmode/grabtern-frontend/assets/120899038/b352548a-da43-4666-8a42-234e58d0c333)
  i have also added discord social icon

ePSA-eJya commented:
@anmode if you could review. i have tried completeing all the tasks.

anmode commented on pages/_document.page.js:
```diff
@@ -35,6 +35,9 @@ export default function Document() {
         <link rel="stylesheet" href="/assets/css/nice-select.css" />
         <link rel="stylesheet" href="/assets/css/style.css" />
 
+        <link rel="stylesheet" href="/assets/css/footer.css" />
+
+
```
ohh no no! create css file in styles folder. see hackathon.css

ePSA-eJya replied:
ok sure

anmode commented on pages/search.js:
```diff
@@ -0,0 +1 @@
+const PAGE_SIZE=2;
```
why this?

ePSA-eJya replied:
i was working on the other issue as well (forgot to remove the folder :p)

anmode replied:
oh np!

anmode commented on public/assets/css/style.css:
```diff
@@ -3736,10 +3738,121 @@ hr {
 }
-.footer-bg {
-  background: #4255a4;
+
+.footer-social ul li {
+	list-style: none;
+}
+
+.footer-social ul li a {
+	position: relative;
+	width: 40px;
+	height: 40px;
+  font-size: 14px;
+	background-color: rgba(128, 136, 255, 0.1);
+	text-align: center;
+	line-height: 40px;
+	border-radius: 50%;
+	overflow: hidden;
+	z-index: 1;
+
+	color: #fff;
+	text-decoration: none;
+	font-size: 20px;
+	flex-basis: 35%;
+	margin-right: 8px;
+	display: inline-block;
+	transition: transform 0.5s;
```
if you have include footer css in a file ? why here to?

ePSA-eJya replied:
ya i will remove those

anmode reviewed (changes requested):
I think it is bit tricky! As footer CSS is present at many places ig. 
so the approach is to see footer js carefully and seek for CSS and include all CSS in a file named footer.css in styles folder.
for refer see : https://github.com/anmode/grabtern-frontend/pull/224

and here is I'm assigning level 3.. Please have patience. it will merged soon ig these would be last changes thank you!.

ePSA-eJya commented:
> I think it is bit tricky! As footer CSS is present at many places ig. so the approach is to see footer js carefully and seek for CSS and include all CSS in a file named footer.css in styles folder. for refer see : #224
> 
> and here is I'm assigning level 3.. Please have patience. it will merged soon ig these would be last changes thank you!.

yes sure....i will try my best to complete it

ePSA-eJya commented:
@anmode i have tried made the footer.module.css file. please review

anmode commented on public/assets/css/style.map:
Just curious? the changes in this file did by you ? oh ok you did changes in style.css and here is map oh ok

ePSA-eJya replied:
yes....very time consuming

anmode commented on public/assets/css/main.css:
Oh! actually this is what i was suggesting.. Please put all the css of footer inside one file.. that you created the footer.css
I know if you remove the css from this file it will affect the footer.. no worries you can see the css and tranfer all into footer.css.

ePSA-eJya replied:
i have already transferred css of footer into foote.modeule.css

anmode reviewed (changes requested):
Looks cool! if you need any help you can ask. thanks

ePSA-eJya commented:
> Screen.Recording.2023-06-04.at.6.27.54.PM.mov 
> and please revert back to original footer... see production color bug: hint1 - Reduce the size of the expanded or make space. hint2- Do not apply many properties and I see many CSS applied same time some CSS properties are not needed. Please check by inspecting I see many CSS properties ,some conflicting ones applied at the same time

i have already transferred all the css of footer into footer.css.module. i think that is what you were expecting???
do you want to revert back to the original social icons?? 
please let me know : ) @anmode

ePSA-eJya commented:
> Screen.Recording.2023-06-04.at.6.27.54.PM.mov 
> and please revert back to original footer... see production color bug: hint1 - Reduce the size of the expanded or make space. hint2- Do not apply many properties and I see many CSS applied same time some CSS properties are not needed. Please check by inspecting I see many CSS properties ,some conflicting ones applied at the same time

i have fixed the issue as shown in the video.

https://github.com/anmode/grabtern-frontend/assets/120899038/b7b19566-5222-4754-9eec-3724f798b90a

anmode commented:
> > Screen.Recording.2023-06-04.at.6.27.54.PM.mov
> > and please revert back to original footer... see production color bug: hint1 - Reduce the size of the expanded or make space. hint2- Do not apply many properties and I see many CSS applied same time some CSS properties are not needed. Please check by inspecting I see many CSS properties ,some conflicting ones applied at the same time
> 
> i have already transferred all the css of footer into footer.css.module. i think that is what you were expecting??? i did not understand what you want me to revert..... please let me know : ) @anmode

ohh I see 
1. you transfer the css but please see in PR there is two css files one is footer.css and other is style.css. can you move all footer css in one file which is footer.css
2. Footer bg color... see production color www.grabtern.com
If there is any query please lemme know. thanks

anmode commented:
> > > Screen.Recording.2023-06-04.at.6.27.54.PM.mov
> > > and please revert back to original footer... see production color bug: hint1 - Reduce the size of the expanded or make space. hint2- Do not apply many properties and I see many CSS applied same time some CSS properties are not needed. Please check by inspecting I see many CSS properties ,some conflicting ones applied at the same time
> > 
> > 
> > i have already transferred all the css of footer into footer.css.module. i think that is what you were expecting??? i did not understand what you want me to revert..... please let me know : ) @anmode
> 
> ohh I see
> 
> 1. you transfer the css but please see in PR there is two css files one is footer.css and other is style.css. can you move all footer css in one file which is footer.css
> 2. Footer bg color... see production color [www.grabtern.com](http://www.grabtern.com). I want you to have that color only that's why I am saying please revert back the bg color
>    If there is any query please lemme know. thanks

ePSA-eJya commented:
> > > Screen.Recording.2023-06-04.at.6.27.54.PM.mov
> > > and please revert back to original footer... see production color bug: hint1 - Reduce the size of the expanded or make space. hint2- Do not apply many properties and I see many CSS applied same time some CSS properties are not needed. Please check by inspecting I see many CSS properties ,some conflicting ones applied at the same time
> > 
> > 
> > i have already transferred all the css of footer into footer.css.module. i think that is what you were expecting??? i did not understand what you want me to revert..... please let me know : ) @anmode
> 
> ohh I see but You did not commit the fix! but carefully look at your video to see that hovering the first icon effect the position of all icons. so the solution is to increase the space in between the icon for a case and remove the Pinterest icon.
> 
> 1. you transfer the CSS but please see in PR there are two CSS files one is footer.css and the other is style.css. can you move all footer css in one file which is footer.css
> 2. Footer bg color... see production color [www.grabtern.com](http://www.grabtern.com). I want you to have that color only that's why I am saying please revert back the bg color
>    If there is any query please lemme know. thanks

i have fixed the social icon link and bg. i will commit that soon.
@anmode i am sorry that i am asking this again and again but i am not really able to understand the css issue. all the footer properties have been transferred from styles.css to footer.module.css. however there is just this->properties__footer which is remaining in style.css. but the problem is that i dont see properties__footer being implement inany of the file.
so is properties__footer necessary? or can be removed.

anmode commented:
I fixed! Congrats @ePSA-eJya

ePSA-eJya commented:
> I fixed! Congrats @ePSA-eJya . I request you to share the pR experience in the group. I know it is bit long hope you enjoyed and learnt

thanks a lot. It was really time taking but I really learnt a lot. thanks for your help @anmode . could you please add the gssoc label on the pr.  thank you
